src/cider/type/plan/BinaryExpr.cpp

```diff
@@ -39,7 +39,9 @@ JITExprValue& BinOper::codegen(JITFunction& func) {
 
   const auto& lhs_ti = lhs->get_type_info();
   const auto& rhs_ti = rhs->get_type_info();
-  CHECK_EQ(lhs_ti.get_type(), rhs_ti.get_type());
+  if (!lhs_ti.is_string()) {
```

Comment:
I think, (though it is not very important), for completeness and type safety, when `lhs_ti` is a string type, perhaps we should type-check that `rhs_ti` is also a string type?